conformance/lang/lang-library/lang.int/abs.balt

```diff
@@ -0,0 +1,533 @@
+
+    testInt(a3.abs()); // @output 20
+
+    testInt(int:abs(a4)); // @output 10
+
+    testInt(a5.abs()); // @output 12
+
+    testInt(int:abs(a5)); // @output 12
+
+    testInt(a6.abs()); // @output 23
+
+    testInt(int:abs(a6)); // @output 23
+}
+
+function testInt(int i) {
+    io:println(i);
+}
+
+Test-Case: output
+Description: Test assignability of `abs` function result with compatible types.
+Labels: int:abs, any, DecimalNumber, function-call-expr, int, method-call-expr, optional-type, string,
+        union-type, unary-minus
+
+function init() {
+    int a1 = -12536;
+
+    int? b1 = a1.abs();
+    io:println(b1); // @output 12536
+
+    int|string b2 = int:abs(a1);
+    io:println(b2); // @output 12536
+
+    testAny(a1.abs());
+}
+
+function testAny(any i) {
+    io:println(i); // @output 12536
+}
+
+Test-Case: output
+Description: Test assignability of `abs` function result with var.
+Labels: int:abs, DecimalNumber, int, is-expr, method-call-expr, unary-minus, var
+
+function init() {
+    int a1 = -12536;
+
+    var b1 = a1.abs();
+    io:println(b1); // @output 12536
+    io:println(b1 is int); // @output true
+}
+
+Test-Case: output
+Description: Test `abs` function result belonging to a type.
+Labels: int:abs, byte, DecimalNumber, function-call-expr, int, int:Signed8, int:Signed16, int:Signed32,
+        int:Unsigned8, int:Unsigned16, int:Unsigned32, is-expr, method-call-expr, unary-minus
+
+function init() {
+    int a1 = -736453762;
+
+    io:println(a1.abs() is int); // @output true
+
+    io:println(int:abs(a1) is int); // @output true
+
+    io:println(-25.abs() is int:Signed8); // @output true
+
+    io:println(-25.abs() is int:Signed16); // @output true
+
+    io:println(int:abs(a1) is int:Signed32); // @output true
+
+    io:println(int:abs(-25) is int:Unsigned8); // @output true
+
+    io:println((-25).abs() is byte); // @output true
+
+    io:println((0xfe).abs() is int:Unsigned16); // @output true
+
+    io:println(a1.abs() is int:Unsigned32); // @output true
+}
+
+Test-Case: output
+Description: Test equality with `abs` function result.
+Labels: int:abs, byte, DecimalNumber, equality, function-call-expr, int, method-call-expr, unary-minus
+
+function init() {
+    int a1 = -25;
+    byte a2 = 25;
+    int a3 = 11100;
+    int:Signed32 a4 = 0x2B5c;
+    int:Signed16 a5 = -0x2B5c;
+
+    int b1 = a1.abs();
+    int b2 = a2.abs();
+
+    io:println(b1 == b2); // @output true
+
+    b1 = int:abs(a1);
+    b2 = int:abs(a2);
+
+    io:println(b1 == b2); // @output true
+
+    io:println(a1.abs() == a2.abs()); // @output true
+
+    io:println(int:abs(a1) == int:abs(a2)); // @output true
+
+    io:println(int:abs(a3) == a5.abs()); // @output true
+
+    io:println(a4.abs() == int:abs(a5)); // @output true
+
+    io:println(a1.abs() != a3.abs()); // @output true
+
+    io:println(int:abs(a2) != int:abs(a3)); // @output true
+}
+
+Test-Case: output
+Description: Test chained `abs` function.
```

Comment:
Do we have tests where we do chained calls with different int langlib methods?

e.g.,

```ballerina
import ballerina/io;

function init() {
    int a = -1;
    int b = a.max(12, 5).min(a, -2).abs();
    io:println(b); // @output 2
}
```

Reply:
These tests are already added. Similar test has been added in `min.balt`

Reply:
Moved